### DISCUSSION TITLE
How to notify (get email) but not ban

### DISCUSSION THREAD

**Discussion by raquelhortab** [fail2ban/fail2ban#3841] [Q&A] [answered]
I have written a custom filter but before I start banning IPs with it I'd like to test it by enabling notifications (emails) without ever banning anyone.

I have tried setting bantime=0 but then I don't get the email. The emails do work since I get them once I set bantime=1.
This is my jail.local:

```
[DEFAULT]
mta=mail
destemail = mymail@gmail.com
action = %(action_mwl)s

[local-example]
enabled=true
port     = http,https
filter = rails-406
logpath = /etc/fail2ban/filter.d/example.log
maxretry = 3
findtime = 60 
bantime = 0
```

When bantime is 0, the status returns this:

```
Status for the jail: local-example
|- Filter
|  |- Currently failed:	1
|  |- Total failed:	4
|  `- File list:	/etc/fail2ban/filter.d/example.log
`- Actions
   |- Currently banned:	0
   |- Total banned:	0
   `- Banned IP list:
```

Is there any way to receive the email without actually banning?

Thanks

**Comment by sebres** [comment, marked as answer]
> I have tried setting bantime=0 but then I don't get the email. 

I don't think it is a bantime (however `bantime = 0` is a bit unexpected, because ban may become immediately obsolete and then the ban may not happen), but even with `bantime = 1`, it can be too short (depending on load etc).

Although ban for few seconds is acceptable time, I think.

> Is there any way to receive the email without actually banning?

Yes, you can rewrite action tag for the jail (the mailing entry only, how it is specified in `action_mwl`, see below):
```ini
[local-example]
enabled = true
...
action = %(mta)s-whois-lines[sender="%(sender)s", dest="%(destemail)s", logpath="%(logpath)s", chain="%(chain)s"]
```
from:
https://github.com/fail2ban/fail2ban/blob/363c0d5fd074abfb394acdea91d547727881ac17/config/jail.conf#L220-L221

Or may be add `action_ =` (to empty value):
```ini
action_ = 
```
(untested)

**Reply by raquelhortab** [reply]
Hi, thanks for looking into it. However, both options you mentioned still ban the address.

**Reply by raquelhortab** [reply]
could you maybe open the discussion again since it has not been solved? @sebres

**Reply by sebres** [reply]
There is nothing to ban, if you would add only 1 action `%(mta)s-whois-lines` (no banning action). Important is to restart the jail or fail2ban hereafter.

**Reply by raquelhortab** [reply]
@sebres  well, the ip gets banned... at least status says so:

```
Status for the jail: local-example
|- Filter
|  |- Currently failed:	1
|  |- Total failed:	5
|  `- File list:	/etc/fail2ban/filter.d/example.log
`- Actions
   |- Currently banned:	1
   |- Total banned:	1
   `- Banned IP list:	94.156.64.12

```
and yes, I restart the service every time

maybe there's some other configuration I got wrong?

```
[DEFAULT]
mta=mail
destemail = xxxxxx@gmail.com
action = %(action_mwl)s
bantime = 30

[rails-406]
enabled  = false
port     = http,https
filter = rails-406
logpath = /home/raquel/RubymineProjects/app/log/development.log
bantime = 3600
maxretry = 3
findtime = 600
action = iptables-multiport[name=Rails-406, port="http,https"]

[local-example]
enabled=true
port     = http,https
filter = rails-406
logpath = /etc/fail2ban/filter.d/example.log
maxretry = 3
findtime = 60 
action = %(mta)s-whois-lines[sender="%(sender)s", dest="%(destemail)s", logpath="%(logpath)s", chain="%(chain)s"]
action_ = 

```

the default action should not affect if I override it right?

**Reply by raquelhortab** [reply]
ok, I think the thing is that it's listed as banned even though it is not effectively banned, so the solution does work, thanks!

**Reply by raquelhortab** [reply]
since I am here, is there any documentation on how to write filters? I did not find anything @sebres

**Reply by sebres** [reply]
[man jail.conf.5](https://manpages.debian.org/experimental/fail2ban/jail.conf.5.en.html)

There also many wikis and how-to's in internet, but also here - https://github.com/fail2ban/fail2ban/wiki/Developing-Regex-in-Fail2ban
